moped-editor/src/views/projects/projectView/ProjectActivityLogDialog.js

```diff
@@ -0,0 +1,323 @@
+
+import { PROJECT_ACTIVITY_LOG_DETAILS } from "../../../queries/project";
+import makeStyles from "@material-ui/core/styles/makeStyles";
+
+const useStyles = makeStyles(theme => ({
+  appBar: {
+    position: "relative",
+  },
+  title: {
+    marginLeft: theme.spacing(2),
+    flex: 1,
+  },
+  paper: {
+    padding: "0",
+  },
+  diffTitle: {
+    padding: "1rem 0",
+  },
+  eventDetail: {
+    padding: "1rem 0",
+    "font-size": "1.5rem",
+  },
+  listColorGray: {
+    color: "gray",
+  },
+  listColorBlack: {
+    color: "black",
+  },
+}));
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+
+const ProjectActivityLogDialog = ({ activity_id, handleClose }) => {
+  const classes = useStyles();
+
+  const { loading, error, data } = useQuery(PROJECT_ACTIVITY_LOG_DETAILS, {
+    variables: {
+      activityId: activity_id,
+    },
+  });
+
+  const [showDiffOnly, setShowDiffOnly] = useState(true);
+
+  const getEventData = (data, mode) => {
+    return data["moped_activity_log"][0]["record_data"]["event"]["data"][mode];
+  };
+
+  const getEventAttribute = (data, field) => {
+    return data?.moped_activity_log[0][field] ?? null;
+  };
+
+  const getUser = data => {
+    try {
+      return (
+        data.moped_activity_log[0].moped_user.first_name +
+        " " +
+        data.moped_activity_log[0].moped_user.last_name
+      );
+    } catch {
+      return null;
+    }
+  };
+
+  const getDateTime = data => {
+    const date = data?.moped_activity_log[0]?.created_at;
+
+    try {
+      return new Date(date).toLocaleString();
+    } catch {
+      return null;
+    }
+  };
+
+  const generateValue = value => {
+    return value === null || String(value).trim() === "" ? (
+      <span className={classes.listColorGray}>Null</span>
+    ) : (
+      <>{value}</>
+    );
+  };
+
+  const generateFieldList = (data, mode) => {
+    const recordState = getEventData(data, mode);
+    const diffFieldList = getEventAttribute(data, "description").map(diff => {
+      return diff.field;
+    });
+    const recordType = getEventAttribute(data, "record_type");
+    const recordTypeFriendly = getRecordTypeLabel(recordType);
+    if (recordState === null) {
+      return null;
+    }
+
+    return (
+      <List dense={false}>
+        {Object.keys(recordState).map(field => {
+          if (showDiffOnly) {
+            if (!diffFieldList.includes(field.toLowerCase())) {
+              return null;
+            }
+          }
+
+          return (
+            <ListItem key={field}>
+              <ListItemAvatar>
+                <Avatar>
+                  <Icon>info</Icon>
+                </Avatar>
+              </ListItemAvatar>
+              <ListItemText
+                primary={
+                  <span
+                    style={{
+                      color: diffFieldList.includes(field.toLowerCase())
+                        ? "red"
```

Comment:
Thank you! You are right, it would be as simple as using two separate style in the class instead of string, I also noticed I did not comment many of the functions in this component. Will have that refactored and merge away! :)